.github/workflows/e2e_nightly_upgrade.yml

```diff
@@ -0,0 +1,204 @@
+    strategy:
+      matrix:
+        ps_start_version: ["1.7.4.4", "1.7.5.2", "1.7.6.9", "1.7.7.1"]
+
+        ps_target_version: ${{ fromJson(needs.get_ps_target_versions.outputs.target_versions) }}
+
+      fail-fast: false
+
+    steps:
+
+      # Setup PrestaShop and Install Autoupgrade module
+      - uses: actions/checkout@v2
+
+      # Build PrestaShop with docker-compose
+      - uses: shivammathur/setup-php@v2
+        name: Setup PHP
+        with:
+          php-version: '7.1'
+          extensions: mbstring, intl, gd, xml, dom, json, fileinfo, curl, zip, iconv
+
+      - name: Install Module dependencies
+        shell: bash
+        run: composer install
+
+      - name: Build PrestaShop ${{ matrix.ps_start_version }} on docker
+        env:
+          VERSION: ${{ matrix.ps_start_version }}
+        shell: bash
+        run : |
+          docker-compose -f docker-compose.yml up -d
+          bash -c 'while [[ "$(curl -L -s -o /dev/null -w %{http_code} http://localhost:8001/index.php)" != "200" ]]; do sleep 5; done'
+
+      # Install Module with user interface
+      - name: Setup Node
+        uses: actions/setup-node@v2
+        with:
+          node-version: 10
+
+      - name: Install dependencies
+        run: npm install
+        working-directory: ./tests/e2e
+
+      - name: Install autoupgrade module
+        run: npm run install-module
+        working-directory: ./tests/e2e
+        env:
+          URL_FO: "http://localhost:8001/"
+          PS_VERSION: ${{ matrix.ps_start_version }}
+
+      - name: Download nightly build
+        run: |
+          docker exec -u www-data prestashop_autoupgrade mkdir -p admin-dev/autoupgrade/download
+          docker exec -u www-data prestashop_autoupgrade curl ${{ matrix.ps_target_version.archive_zip }} -o admin-dev/autoupgrade/download/${{ matrix.ps_target_version.filename }}
+          docker exec -u root prestashop_autoupgrade chmod 777 -R /var/www/html
+
+      # Run Upgrade tests
+      - name: Run tests
+        run: npm run upgrade-test
+        working-directory: ./tests/e2e
+        env:
+          URL_FO: "http://localhost:8001/"
+          PS_VERSION: ${{ matrix.ps_start_version }}
+          PS_VERSION_UPGRADE_TO: ${{ matrix.ps_target_version.for_test }}
+          ZIP_NAME: ${{ matrix.ps_target_version.filename }}
+
+      - uses: actions/upload-artifact@v2
+        name: Upload failed screenshots
+        if: ${{ always() }}
+        with:
+          name: failed-steps-for-test-${{ matrix.ps_start_version }}-${{ matrix.ps_target_version.for_test }}
+          path: ./tests/e2e/screenshots/
+
+      # Upload Mochawesome reports
+      - name: Rename Mochawesome report
+        if: ${{ always() }}
+        run: |
+          mkdir -p ./json-reports
+          cp ./tests/e2e/mochawesome-report/mochawesome.json ./json-reports/upgrade-from-${{ matrix.ps_start_version }}-to-${{ matrix.ps_target_version.for_test }}.json
```

Comment:
A lot of tasks should be in dedicated scripts instead, it will be easier to manage, use and reproduce (as it's done in the PrestaShop Core repository)

Reply:
done